receiver/zipkinreceiver/trace_receiver.go

```diff
@@ -217,7 +219,8 @@ func (zr *ZipkinReceiver) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	transportTag := transportType(r, asZipkinv1)
 	ctx = obsreport.ReceiverContext(ctx, zr.id, transportTag)
-	ctx = obsreport.StartTraceDataReceiveOp(ctx, zr.id, transportTag)
+	zr.receiver = obsreport.NewReceiver(obsreport.ReceiverSettings{ReceiverID: zr.id, Transport: transportTag})
```

Comment:
Here you cannot "cache" since you don't know the transport (actually it will cause a race condition). you cannot do things blindly :) Please check everywhere if that is the case.

Reply:
Ohhh ok I thought so.  I just fixed this in the latest commit.  I think that was the only place where we do not know the transport.